pkg/backend/diy/postgres/bucket.go

```diff
@@ -0,0 +1,439 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"gocloud.dev/blob"
+	"gocloud.dev/blob/driver"
+	"gocloud.dev/gcerrors"
+)
+
+// PostgresBucket implements blob.Bucket storage using PostgreSQL.
+type PostgresBucket struct {
+	db        *sql.DB
+	tableName string
+	bucket    *blob.Bucket
+}
+
+// tableSchema defines the SQL schema for the table that will store the blobs.
+const tableSchema = `
+CREATE TABLE IF NOT EXISTS %s (
+    key TEXT PRIMARY KEY,
+    data BYTEA NOT NULL,
```

Comment:
Could this be a `json` field instead?  That would give the user a few more options, and I *think* we always store json data in here.